source/Halibut/Transport/PollingClient.cs

```diff
@@ -40,14 +41,23 @@ public void Start()
 
         private void ExecutePollingLoop(object ignored)
         {
+            var retry = new PhasedBackoffRetryTracker();
             while (working)
             {
                 try
                 {
+                    retry.Try();
                     secureClient.ExecuteTransaction(protocol =>
                     {
                         protocol.ExchangeAsSubscriber(subscription, handleIncomingRequest);
                     });
+                    retry.Success();
+                }
+                catch (HalibutClientException hce)
```

Comment:
I was wrong, the code does not fall through to this block regularly. However it does so with a `HalibutClientException`. The next catch is for those really out of ordinary cases.